src/lib.rs

```diff
@@ -12,6 +12,95 @@ extern crate ethereum_types;
 #[cfg(test)]
 extern crate serde_json;
 
-mod raw_transaction;
+use secp256k1::{SecretKey, Message, Secp256k1};
+use rlp::RlpStream;
+use rlp::Encodable;
 
+mod raw_transaction;
 pub use self::raw_transaction::RawTransaction;
+
+pub trait Transaction {
+    fn chain_id(&self) -> u64;
+    fn rlp(&self) -> &[&dyn Encodable];
```

Comment:
Is this the generic function to encode the transaction? Maybe we should rename it to `encode()`, just since EIP2718 says that the encoding can be any serialization algorithm, not limited to RLP.